riscv/decode.h

```diff
@@ -2836,13 +2836,12 @@ for (reg_t i = 0; i < P.VU.vlmax && P.VU.vl != 0; ++i) { \
 #define P_PK(BIT, X, Y) \
   require_extension('P'); \
   require(BIT == e16 || BIT == e32); \
-  reg_t size = BIT / 8 * 2; \
-  reg_t rd_tmp = 0; \
-  type_usew_t<BIT>::type pd[2] = { \
-    P_UFIELD(RS1, Y, BIT), \
-    P_UFIELD(RS1, X, BIT) \
-  }; \
-  memcpy(&rd_tmp, pd, size); \
+  reg_t rd_tmp = 0, rs1 = RS1, rs2 = RS2; \
```

Comment:
@marcfedorow After I merged your PR, I found that your commit missed the for expression.
Can you solve this and create new PR?

Reply:
Solved